ament_package/template/isolated_prefix_level/local_setup.bat.in

```diff
@@ -24,14 +24,22 @@ goto:eof
   setlocal enabledelayedexpansion
 
   :: use the Python executable known at configure time
-  set "ament_python_executable=@PYTHON_EXECUTABLE@"
+  set "_ament_python_executable=@PYTHON_EXECUTABLE@"
   :: allow overriding it with a custom location
   if "%AMENT_PYTHON_EXECUTABLE%" NEQ "" (
-    set "ament_python_executable=%AMENT_PYTHON_EXECUTABLE%"
+    set "_ament_python_executable=%AMENT_PYTHON_EXECUTABLE%"
+  )
+  if NOT exist %_ament_python_executable% (
+    for %%I in (python.exe) do set "exepath=%%~$PATH:I"
+    if "!exepath!" NEQ "" set "_ament_python_executable=!exepath!"
```

Comment:
Yes, but then there needs to be temporary variable in the error condition. Setting the variable directly opens up the possibility that it is empty. If it is empty, the error condition needs to implement an "or" operation with a temporary variable: http://stackoverflow.com/questions/2143187/logical-operators-and-or-in-dos-batch.